File: ios/EnterpriseMobileApp/Sources/Core/Domain/Repositories/CartRepository.swift
import Foundation

/// Repository protocol for shopping cart operations.
public protocol CartRepository: Sendable {
    func getCartItems() -> AsyncStream<[CartItem]>
    func addItem(_ item: CartItem) async -> DomainResult<Void>
    func removeItem(id: String) async -> DomainResult<Void>
    func updateQuantity(itemId: String, quantity: Int) async -> DomainResult<Void>
    func clearCart() async -> DomainResult<Void>
    func getCartItemCount() async -> Int
}

File: android/core/domain/src/main/java/com/enterprise/core/domain/repositories/CartRepository.kt
package com.enterprise.core.domain.repositories

import com.enterprise.core.domain.entities.CartItem
import com.enterprise.core.domain.entities.DomainResult
import kotlinx.coroutines.flow.Flow

/** Shopping cart repository contract with reactive item stream. */
interface CartRepository {
    fun getCartItems(): Flow<List<CartItem>>
    suspend fun addItem(item: CartItem): DomainResult<Unit>
    suspend fun removeItem(id: String): DomainResult<Unit>
    suspend fun updateQuantity(itemId: String, quantity: Int): DomainResult<Unit>
    suspend fun clearCart(): DomainResult<Unit>
    suspend fun getCartItemCount(): Int
}
